File: examples/php/basic.php
<?php
/**
 * Basic oruk usage: public feed → authed filtered list → pagination.
 *
 *   export ORUK_API_KEY=ork_xxxxxxxx
 *   php basic.php
 */
declare(strict_types=1);

require __DIR__ . '/client.php';

$oruk = new OrukClient(getenv('ORUK_API_KEY') ?: null);

// ----- 1. Public feed (no key required) -----
echo "=== /v1/stories/feed (public) ===\n";
$feed = $oruk->feed(['limit' => 5, 'sort' => 'recent']);
foreach ($feed['stories'] as $s) {
    printf(
        "[%s] %-9s %s\n  %s\n",
        $s['id'],
        $s['urgency'] ?? '?',
        $s['headline'] ?? '(no headline)',
        '   sources=' . ($s['corroboration']['count'] ?? 0)
            . ' impact='  . ($s['impact'] ?? '?')
            . ' city='    . ($s['eventCity'] ?? '?')
    );
}

// Everything below this point requires a key.
if (!getenv('ORUK_API_KEY')) {
    echo "\nSet ORUK_API_KEY to see the authed examples.\n";
    exit(0);
}

// ----- 2. Authed filtered list -----
echo "\n=== /v1/stories?category=conflict&min_impact=7 ===\n";
$conflict = $oruk->stories([
    'category'   => 'conflict',
    'min_impact' => 7,
    'limit'      => 5,
]);
foreach ($conflict['stories'] as $s) {
    printf("[%2d] %s\n", $s['impact'] ?? 0, $s['headline'] ?? '');
}

// ----- 3. Pagination via cursor -----
echo "\n=== Paging through last 7 days of economy stories (max 25) ===\n";
$since = gmdate('Y-m-d\TH:i:s\Z', strtotime('-7 days'));
$count = 0;
foreach ($oruk->streamPages(['category' => 'economy', 'since' => $since, 'limit' => 50]) as $s) {
    $count++;
    if ($count <= 5) {
        printf("  %s\n", $s['headline'] ?? '');
    }
    if ($count >= 25) break;
}
printf("\nPulled %d stories across pages.\n", $count);

// ----- 4. Stats -----
echo "\n=== /v1/stats ===\n";
$stats = $oruk->stats();
printf("Active sources: %d\nStories total:  %d\n",
    $stats['activeSources'] ?? 0,
    $stats['storiesTotal']  ?? 0,
);
echo "Top categories:\n";
foreach (array_slice($stats['topCategories'] ?? [], 0, 5) as $c) {
    printf("  %-12s %d\n", $c['category'] ?? '?', $c['count'] ?? 0);
}

File: examples/ruby/basic.rb
#!/usr/bin/env ruby
# Basic oruk usage: public feed -> authed list -> pagination.
#
#   export ORUK_API_KEY=ork_xxxxxxxx
#   ruby basic.rb
require_relative 'oruk_client'

oruk = OrukClient.new(ENV['ORUK_API_KEY'])

puts '=== /v1/stories/feed (public) ==='
oruk.feed(limit: 5, sort: 'recent')['stories'].each do |s|
  puts "[#{s['id']}] #{(s['urgency'] || '?').ljust(9)} #{s['headline']}"
  puts "   sources=#{s.dig('corroboration','count') || 0} impact=#{s['impact'] || '?'} city=#{s['eventCity'] || '?'}"
end

unless ENV['ORUK_API_KEY']
  puts; puts 'Set ORUK_API_KEY to see the authed examples.'; exit
end

puts; puts '=== /v1/stories?category=conflict&min_impact=7 ==='
oruk.stories(category: 'conflict', min_impact: 7, limit: 5)['stories'].each do |s|
  puts "[#{(s['impact'] || 0).to_s.rjust(2)}] #{s['headline']}"
end

puts; puts '=== Paging through last 7 days of economy stories (max 25) ==='
since = (Time.now.utc - 7 * 24 * 3600).strftime('%Y-%m-%dT%H:%M:%SZ')
count = 0
oruk.each_page(category: 'economy', since: since, limit: 50) do |s|
  count += 1
  puts "  #{s['headline']}" if count <= 5
  break if count >= 25
end
puts; puts "Pulled #{count} stories across pages."

puts; puts '=== /v1/stats ==='
stats = oruk.stats
puts "Active sources: #{stats['activeSources'] || 0}"
puts "Stories total:  #{stats['storiesTotal'] || 0}"
puts 'Top categories:'
(stats['topCategories'] || []).first(5).each do |c|
  puts "  #{(c['category'] || '?').ljust(12)} #{c['count'] || 0}"
end
